apis/v1beta1/validation/httproute.go

```diff
@@ -348,6 +353,21 @@ func validateHTTPHeaderModifier(filter gatewayv1b1.HTTPHeaderFilter, path *field
 	return errs
 }
 
+func validateHTTPRouteTimeouts(timeouts *gatewayv1b1.HTTPRouteTimeouts, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if timeouts.BackendRequest != nil {
+		backendTimeout, _ := time.ParseDuration((string)(*timeouts.BackendRequest))
+		if timeouts.Request != nil {
+			timeout, _ := time.ParseDuration((string)(*timeouts.Request))
+			if backendTimeout > timeout {
+				errs = append(errs, field.Invalid(path.Child("backendRequest"), backendTimeout, "backendRequest timeout cannot be longer than request timeout"))
```

Comment:
If I'm reading this right - and I may not be - then

```yaml
timeouts:
    request: 0s
    backendRequest: 100ms
```

will fail this test, even though that's a valid timeout configuration. Am I reading this correctly?

Reply:
good catch, I had mistakenly removed the `&& timeout != 0` on one of the last commits - re-adding that now. thank @kflynn

Reply:
addressed in d8d74726cc2f2a1c2e1e853865745ce3caee3af4

Reply:
@SRodi I thought there was a unit test for this case, but I see it's just in the CEL validation test. Please make sure that all the CEL tests are also replicated in the validator tests.